src/v/net/conn_quota.h

```diff
@@ -238,6 +236,7 @@ class conn_quota : public ss::peering_sharded_service<conn_quota> {
     conn_quota_config _cfg;
 
     ss::gate _gate;
+    seastar::logger* _log;
```

Comment:
nit: If `_log` is non-nullable and won't be reseated, prefer a reference over a pointer.

Reply:
clang-tidy really doesn't like member references. they are a bit weird. wdyt @rockwotj is there a non-null thingy in abseil for this?

Reply:
I personally prefer pointers for a few reasons I can elaborate on, but yes, absl has helpers but we have to update absl: https://github.com/abseil/abseil-cpp/blob/master/absl/base/nullability.h

Reply:
>I personally prefer pointers for a few reasons I can elaborate on

Same, but it's more of a vibe. So I'm curious about your throughts. Maybe we should update the coding-style.md with some reasoning?

Reply:
I'll open a thread in #cpplang